aip/general/0162.md

```diff
@@ -25,201 +25,170 @@ to the version of an API as a whole.
+- Create a new revision when the user specifically asks
 
-    publishers/123/books/les-miserables@c7cfa2a8
+APIs **may** use any of these strategies. If a strategy is used, they **must**
+document their revision creation strategy.
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+### Resource names for revisions
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+When referring to specific revision of a resource, the subcollection name
+**must** be named `revisions`. Resource revisions have names with the format
+`{resource_name}/revisions/{revision_id}`. For example:
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+publishers/123/books/les-miserables/revisions/c7cfa2a8
 
-### Getting a revision
+### Server specified IDs
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+Services **may** reserve specific IDs to be server-specified (e.g. `current`,
+`latest`), which are read-only and managed by the service.
 
-```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
-  string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
-    }];
-}
+```
+GET /v1/publishers/{publisher}/books/{book}/revisions/{revision_id}
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- If a `latest` ID exists, it **must** represent the most recently created
+revision. The content of `publishers/{publisher}/books/{book}/revisions/latest`
+and `publishers/{publisher}/books/{book}` can differ, as the latest revision may
+be different from the current state of the resource.
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+### Tagging
 
-### Tagging revisions
+Rather than create a revision via the standard method, it may be more convenient
```

Comment:
It feels like we're currently in a bit of a half-way house between referring to tags and not doing so. At the moment we have both concepts not just in the description, but in code, via the `alternate_ids` field and the `TagXyzRevision` custom method.

I don't think this is blocking, but I always get nervous when we refer to basically the same thing with two different terms - that's tended to work out badly over time, in my experience. If we can justify it, that's fine.

Reply:
I do agree with nomenclature discrepancies (alternate_ids vs tags). 

The way I was looking at this was "tagging" is an action that you can do with revisions, but there is no "tag" object: it's just a custom method for convenience.

Do you think that's not clear from this AIP? or maybe a different suggestion?

ack on this being non-blocking, just always a good time to have the discussion while we both have context.

Reply:
This is more about terminology than actual resources.

If we want to use tag/tagging only as a verb, this should be rewritten:

> This allows users to write code against specific tags (e.g. published) and the revision can change in the background with no code change.

We might also want to make it really clear in the "Using resource ID instead of tag" - explicitly say that this is what `alternate_ids` (or `aliases` if we rename it) is for.